src/core/exec-invoke.c

```diff
@@ -4904,7 +4904,7 @@ static void prepare_terminal(
 
```

Comment:
A few lines above we have
```C
        /* We only try to reset things if we there's the chance our stdout points to a TTY */
        if (!(is_terminal_output(context->std_output) ||
              (context->std_output == EXEC_OUTPUT_INHERIT && is_terminal_input(context->std_input)) ||
              context->std_output == EXEC_OUTPUT_NAMED_FD ||
              p->stdout_fd >= 0))
                return;
```
I find that code quite surprising. If we're writing to a named fd, then it's a socket or a fifo or a file. Why would we do any of the subsequent steps in that case? This code was added like this originally and I don't understand the logic.

@poettering ?